lapce-ui/src/hover.rs

```diff
@@ -177,75 +179,24 @@ impl Widget<LapceTabData> for HoverContainer {
     }
 }
 
-enum HoverLayout {
-    Text {
-        text_layout: PietTextLayout,
-        /// The y position including this layout
-        y: f64,
-    },
-    /// Empty, should be ignored
-    Empty,
-    /// The final entry
-    Final { height: f64 },
-}
-
 #[derive(Default)]
-pub struct Hover {}
+pub struct Hover {
+    /// The active text layout to be rendered
+    /// Uses [`RichText`] to unify non-markdown and markdown text together
+    active_layout: TextLayout<MarkdownText>,
```

Comment:
```Uses [`RichText`] to unify non-markdown and markdown text together``` seems incorrect here, just by looking at these two lines. Could you clarify that you mean this refers to inside `MarkdownText`?

Reply:
Ah, I'll fix that. I originally was using `TextLayout<RichText>` instead of `TextLayout<MarkdownText>` but then I realized I needed to keep track of the code block indices so needed a custom time.  
(It is unifying in the sense that I don't have a `TextLayout<String>` and a `TextLayout<MarkdownText>` as an enum or something, I just treat plaintext as unstyled rich/markdown text)